src/Compilers/Core/Portable/StrongName/StrongNameKeys.cs

```diff
@@ -88,14 +88,13 @@ internal static StrongNameKeys Create(ImmutableArray<byte> publicKey, CommonMess
 
         internal static StrongNameKeys Create(string keyFilePath, CommonMessageProvider messageProvider)
         {
-            if (string.IsNullOrEmpty(keyFilePath))
+            if (string.IsNullOrEmpty(keyFilePath) || !PathUtilities.IsAbsolute(keyFilePath))
```

Comment:
I gave that a shot: the difficulty is that the diagnostic about the compilation options doesn't prevent compilation from continuing, so the `PublicSignWithRelativeKeyPath` test fails with a debug assert instead of a diagnostic. Perhaps I need to check the diagnostics from `VerifyOptions` before proceeding with compilation? I was worried about existing behavior relying on compilation continuing.